lib/spack/docs/tutorial_buildsystems.rst

```diff
@@ -0,0 +1,498 @@
+   :emphasize-lines: 45,49,50
+
+Now we can control our build options using :code:`cmake_args()`. If defaults are
+sufficient enough for the package, we can leave out :code:`cmake_args()` 
+in our package.py
+
+:code:`CMakePackage` classes allow for control of other features in the 
+build system. For example, you can specify the path to the "out of source" 
+build directory and also point to the root of the CMakeLists.txt file if it 
+is placed in a non-standard location.
+
+--------------
+PythonPackage
+--------------
+
+Python extensions and modules are built differently from source than most
+applications. Python uses a setup.py script to install Python modules. 
+The script consists of a call to setup() which provides the information 
+required to build a module to Distutils. If you're familiar with pip or 
+easy_install, setup.py does the same thing.
+
+These modules are usually installed using the following line:
+
+.. code-block:: console
+    
+    $ python setup.py install <module>
+
+There are also a list of commands and phases that you can call. To see the full
+list you can run:
+    
+.. code-block:: console
+    
+    $ python setup.py --help-commands
+    Standard commands:
+        build             build everything needed to install
+        build_py          "build" pure Python modules (copy to build directory)
+        build_ext         build C/C++ extensions (compile/link to build directory)
+        build_clib        build C/C++ libraries used by Python extensions
+        build_scripts     "build" scripts (copy and fixup #! line)
+        clean             (no description available)
+        install           install everything from build directory
+        install_lib       install all Python modules (extensions and pure Python)
+        install_headers   install C/C++ header files
+        install_scripts   install scripts (Python or otherwise)
+        install_data      install data files
+        sdist             create a source distribution (tarball, zip file, etc.)
+        register          register the distribution with the Python package index
+        bdist             create a built (binary) distribution
+        bdist_dumb        create a "dumb" built distribution
+        bdist_rpm         create an RPM distribution
+        bdist_wininst     create an executable installer for MS Windows
+        upload            upload binary package to PyPI
+        check             perform some checks on the package
+
+
+For each of these commands, Spack has a method that can be overridden. 
+
+To see the defaults that Spack has for each a methods, we will take a look
+at the :code:`PythonPackage` class:
+
+.. literalinclude:: tutorial/examples/PyPackage/python_package_class.py
+   :language: python
+   :lines: 35, 161-364
+   :linenos:
+
+As noted earlier, each of those methods have sensible defaults or they can
+be overridden.
+
+Using the :code:`Package` class is possible but it brings with it a lot of unnecessary
```

Comment:
Minor edit: This sentence is in passive voice, try

"We can write package files for Python packages using the Package class, but it brings with it a lot methods that are useless for Python packages.